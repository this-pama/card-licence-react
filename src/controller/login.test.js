import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import sinon from 'sinon';
import LoginApp from './login';

describe('test login App page', () => {
  it('renders without crashing', () => {
    shallow(<LoginApp />);
  });
});

describe('login app snapshot', () => {
  it('test snapshot', () => {
    const component = renderer.create(<LoginApp />);

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.handleEmail;
    tree.props.handlePassword;
    tree.props.validateForm;
    tree.props.logIn;
    tree.props.email;
    tree.props.password;
    tree.props.spinner;
    tree.props.errMessage;
    tree.props.disable;

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('simulate events', () => {
  const wrapper = mount(<LoginApp />);

  it('simulates click events', () => {
    const onButtonClick = sinon.spy();
    const wrapper = mount(<LoginApp logIn={onButtonClick} />);
    wrapper.find('Button').simulate('click');
    // expect(onButtonClick).toHaveReturnedWith('errMessage', '');
  });
});

describe('Test internal states of methods', () => {
  const component = renderer.create(<LoginApp />);
  const instance = component.getInstance();

  test('it change state', () => {
    expect(instance.state.password).toBe('');
    expect(instance.state.email).toBe('');
    expect(instance.state.errMessage).toBe('');
    expect(instance.state.spin).toBe(false);
    const value = { target: { value: 'foo' } };
    instance.handleEmail(value);
    expect(instance.state.email).toBe('foo');
    instance.handlePassword(value);
    expect(instance.state.password).toBe('foo');
  });

  test('validate form state', () => {
    instance.validateForm();
    expect(instance.state.disable).toBe(true);
  });

  test('spinner state', () => {
    instance.logIn();
    expect(instance.state.spin).toBe(true);
  });
});
