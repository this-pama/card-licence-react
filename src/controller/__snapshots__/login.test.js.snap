// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login app snapshot test snapshot 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-md-3 col-sm-2"
    />
    <div
      className="col-md-6 col-sm-8"
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginTop": 50,
        }
      }
    >
      <div>
        <div
          style={
            Object {
              "backgroundColor": "blue",
              "border": 1,
              "color": "white",
            }
          }
        >
          <h3
            style={
              Object {
                "padding": 15,
                "textAlign": "center",
              }
            }
          >
            Card Licencing
          </h3>
        </div>
        <p
          style={
            Object {
              "color": "red",
              "paddingTop": 10,
            }
          }
        >
          
        </p>
        <div
          className="form-group"
        >
          <label
            className="form-label"
            htmlFor="formBasicEmail"
          >
            Email address
          </label>
          <input
            className="testEmail form-control"
            id="formBasicEmail"
            onChange={[Function]}
            placeholder="Enter email"
            type="email"
            value=""
          />
          <small
            className="text-muted form-text"
          >
            We'll never share your email with anyone else.
          </small>
        </div>
        <div
          className="form-group"
        >
          <label
            className="form-label"
            htmlFor="formBasicPassword"
          >
            Password
          </label>
          <input
            className="form-control"
            id="formBasicPassword"
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value=""
          />
        </div>
        <button
          className="btn btn-primary btn-block"
          disabled={true}
          onClick={[Function]}
          type="submit"
        >
            Login
        </button>
      </div>
    </div>
    <div
      className="col-md-3 col-sm-2"
    />
  </div>
</div>
`;

exports[`login app snapshot test snapshot 2`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-md-3 col-sm-2"
    />
    <div
      className="col-md-6 col-sm-8"
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginTop": 50,
        }
      }
    >
      <div>
        <div
          style={
            Object {
              "backgroundColor": "blue",
              "border": 1,
              "color": "white",
            }
          }
        >
          <h3
            style={
              Object {
                "padding": 15,
                "textAlign": "center",
              }
            }
          >
            Card Licencing
          </h3>
        </div>
        <p
          style={
            Object {
              "color": "red",
              "paddingTop": 10,
            }
          }
        >
          
        </p>
        <div
          className="form-group"
        >
          <label
            className="form-label"
            htmlFor="formBasicEmail"
          >
            Email address
          </label>
          <input
            className="testEmail form-control"
            id="formBasicEmail"
            onChange={[Function]}
            placeholder="Enter email"
            type="email"
            value=""
          />
          <small
            className="text-muted form-text"
          >
            We'll never share your email with anyone else.
          </small>
        </div>
        <div
          className="form-group"
        >
          <label
            className="form-label"
            htmlFor="formBasicPassword"
          >
            Password
          </label>
          <input
            className="form-control"
            id="formBasicPassword"
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value=""
          />
        </div>
        <button
          className="btn btn-primary btn-block"
          disabled={true}
          onClick={[Function]}
          type="submit"
        >
            Login
        </button>
      </div>
    </div>
    <div
      className="col-md-3 col-sm-2"
    />
  </div>
</div>
`;
